### PR TITLE
[agent-a][issue #594] Define selected-contract fork execution admission

### LINKED ISSUE #594: [runtime feature] Define selected-contract fork execution admission from durable binding
## Summary

Extracted from the approved #588 gate after #590 / PR #592 landed.

The repo now has durable selected-contract source binding for materialized timestamp forks via `store.run_fork.selected_contract_binding`, but future selected-contract execution still has no runtime-side admission/context owner that consumes that binding. This child defines that owner before any mutating handler execution, fork-local event/delivery writes, receipts, dead letters, idempotency, emitted follow-ups, timers, routes, sessions, turns, source-advanced policy, contract-swap execution, or UI/API enablement.

## Category

High-risk runtime/store semantic prerequisite under #588.

This issue is approved as the first slice by the #588 independent gate. Coding may start only inside the boundary below. If implementation needs mutation or delivery/outcome ownership, stop and repair scope under #588 / `delivery_and_replay_ownership`.

## Parent / tracker context

- Parent: #588 mutating selected-contract fork execution owner.
- Broader parent: #564 historical replay/resume ownership.
- Broader architecture parent: #549 mutating timestamp-fork architecture.
- Closed prerequisite: #590 / PR #592 durable selected-contract fork source binding.
- Watchlist mapping: `timestamp_fork_replay_resume_ownership`.

## Working failure class

`runtime.run_fork.selected_contract_execution_admission` is missing.

Future selected-contract execution needs a canonical runtime-side admission/context owner that:

- reads `store.run_fork.selected_contract_binding` by fork `run_id`;
- loads and validates the selected semantic source from the durable binding;
- combines that binding with existing `runtime.run_fork.contract_frontier_admission` and `runtime.run_fork.selected_contract_execution_model` evidence;
- emits a non-mutating execution admission/context result for the later `runtime.run_fork.selected_contract_execution` owner;
- fails closed before mutation when binding/source/admission evidence is missing, ambiguous, stale, incompatible, or unavailable.

## In scope

- Add a named canonical owner such as `runtime.run_fork.selected_contract_execution_admission`.
- Consume `store.run_fork.selected_contract_binding`; do not consume CLI/API args as execution truth.
- Load/validate selected contract source from the binding identity.
- Produce a fail-closed, non-mutating admission/context model for future execution.
- Preserve or explicitly report existing blockers for #573/#575/timers/routes/source-advanced/contract-swap/UI.
- Update `docs/watchlists/runtime-operations.yaml` to map this issue under `timestamp_fork_replay_resume_ownership`.
- Add focused tests for binding read, missing/unavailable selected source, canonical owner accounting, and negative mutation proof.

## Out of scope

- Handler execution.
- Fork-local `events` or `event_deliveries` writes.
- Source event or source `event_deliveries` copying.
- Same-run outbox replay as timestamp-fork replay.
- Receipts, dead letters, idempotency, emitted follow-up events.
- Timer, route, session, turn, source-advanced, contract-swap, Builder/dashboard/API mutation or enablement.
- Reclassifying #570/#573/#575 blockers.

## Required implementation proof

The PR proof audit must include:

- exact spec sections cited from fork materialization/admission and selected-contract execution model;
- canonical owner accounting for selected binding, selected frontier admission, execution model, and this new admission/context owner;
- repo-wide consumer accounting showing CLI/API args, dry-run JSON, route/frontier helpers, activation-local SQL, source row copy, and same-run outbox replay are not execution owners;
- manifestation table with each row classified as fixed through this admission path, split/escalated, or invalid path;
- supported-surface proof that mutating `--contracts` remains blocked and no executable work is created;
- tests proving missing binding/source fails closed and no handler/event/delivery/outcome mutation occurs;
- closure claim limited to this first-slice failure class.

## Stop conditions

Stop and return to #588 if:

- the implementation needs to run handlers;
- the implementation needs to create fork-local event/delivery rows;
- source events/deliveries/outcomes are used as executable fork work or suppressors;
- CLI/API args become durable execution truth;
- normal runtime bus/pipeline/receipt/dead-letter/idempotency owners need to move in the same PR.

## Gate outcome inherited from #588

Approved as first slice.

This issue should close only `runtime.run_fork.selected_contract_execution_admission`. #588 remains open for full mutating selected-contract fork execution.

### PR DESCRIPTION
**Pull request by yazzaoui**
## Summary

Closes #594.

Part of #588.

This PR implements the approved first slice for selected-contract fork execution admission. It adds the non-mutating `runtime.run_fork.selected_contract_execution_admission` owner, consumes `store.run_fork.selected_contract_binding` through a fork-run binding reader, validates the selected semantic source against that durable binding, and combines the binding with existing frontier/model evidence without enabling handler execution or fork-local event/delivery/outcome mutation.

## Governing Context

Spec references:
- `platform_tables.run_fork_selected_contract_bindings`
- `run_model.fork.materialize_only_admission`
- `run_model.fork.activation_admission`
- `run_model.fork.replay_resume_admission_taxonomy`
- `run_model.fork.selected_contract_execution_model`
- New/updated: `run_model.fork.selected_contract_execution_admission`

Binding issue/gate context:
- #588 refreshed Pre-Implementation Coverage Audit
- #588 independent lead gate outcome: approved as first slice
- #594 issue body and approved boundary
- #590 / PR #592 durable selected-contract binding prerequisite

## Semantic Migration / Owner Accounting

New canonical owner:
- `runtime.run_fork.selected_contract_execution_admission`

Existing prerequisite owners consumed:
- `store.run_fork.selected_contract_binding`
- `runtime.run_fork.contract_frontier_admission`
- `runtime.run_fork.selected_contract_execution_model`

Old non-authoritative paths now invalid / still invalid:
- CLI/API `--contracts` arguments are not execution truth.
- Dry-run JSON and local model output are evidence only.
- Route/frontier helpers are not execution owners.
- Source events, source `event_deliveries`, source outcomes, and same-run outbox replay are not selected-contract fork execution.

Production paths checked for surviving old behavior:
- `cmd/swarm fork --contracts` remains non-executing outside dry-run/materialize-only binding.
- `MaterializeRunFork` remains state/binding materialization only.
- `ActivateRunFork` remains activation/source-freeze only.
- `runtime.run_fork.selected_contract_execution` remains the future mutating owner and is not implemented here.

Migration completeness:
- Complete for the approved #594 non-mutating admission/context slice.
- Full mutating selected-contract fork execution remains open under #588.

## Proofs

Focused:
- `go test ./internal/runtime/runforkexecution -count=1`
- `go test ./cmd/swarm -run 'TestRunForkCommand_(ContractsRemainNonExecuting|MaterializeOnlyUsesCanonicalStoreOwnerJSON|ActivateUsesCanonicalStoreOwnerJSON|DryRunContractsAddsContractFrontierAdmissionJSON)' -count=1`
- `go test ./internal/store -run 'TestRunForkSelectedContractBinding|TestRunForkMaterializer_CreatesPausedForkRunAndSnapshotWithoutResuming|TestGeneratePlatformTableDDLs' -count=1`

Relevant/full:
- `go test ./internal/runtime/... -count=1`
- `go test ./internal/store -count=1`
- `go test ./cmd/swarm -count=1`
- `go test ./... -count=1`

### CODE REVIEW

**Comment by yazzaoui** [comment]
## Post-Implementation Proof Audit

PR: #595  
Issue: #594  
Parent: #588  
Commit: `4bd53b7422bf2663eda73fc39de827d5d4610294`

### Semantic Concepts Changed

- Added non-mutating selected-contract fork execution admission/context ownership.
- Added the canonical owner string and DTO for `runtime.run_fork.selected_contract_execution_admission`.
- Made the selected-contract fork stream consume `store.run_fork.selected_contract_binding` before any future execution mutation.
- Updated governing spec/watchlist text for the new admission owner.

### Chosen Working Failure Class

Chosen class closed by this PR:

`runtime.run_fork.selected_contract_execution_admission` was missing.

This PR eliminates that first-slice class by adding a fail-closed, non-mutating admission builder that reads the durable binding by fork `run_id`, validates selected source identity, consumes canonical frontier/model evidence, and returns admission/context for the future mutating owner.

### Parent Failure Class

Immediate parent remains open:

- #588 mutating selected-contract timestamp-fork execution owner.

Broader parents remain open:

- #564 historical replay/resume ownership.
- #549 full mutating timestamp-fork architecture.

### Canonical Owners After Change

| Concept | Canonical owner | Status |
| --- | --- | --- |
| selected-source binding | `store.run_fork.selected_contract_binding` | prerequisite consumed |
| selected frontier evidence | `runtime.run_fork.contract_frontier_admission` | prerequisite consumed |
| selected execution model | `runtime.run_fork.selected_contract_execution_model` | prerequisite consumed |
| selected execution admission/context | `runtime.run_fork.selected_contract_execution_admission` | added/closed |
| future mutating execution | `runtime.run_fork.selected_contract_execution` | still future/split |
| safe pending-agent replay | `store.run_fork.delivery_event_replay` | separate prerequisite/sibling |
| normal pipeline/bus/outcomes | `internal/runtime/pipeline`, `internal/runtime/bus`, `internal/store/event_receipt_store.go`, dead-letter owners | future consumers only |

### Touched Consumers / Systematic Consumption Audit

| Seam | Classification |
| --- | --- |
| `BuildSelectedContractExecutionAdmission` | moved to canonical selected binding owner through `RequireRunForkSelectedContractBinding(ctx, forkRunID)` |
| selected semantic source | validated against durable binding workflow name/version |
| frontier admission | must already be `runtime.run_fork.contract_frontier_admission`, non-mutating, historical execution unsupported |
| execution model | must already be `runtime.run_fork.selected_contract_execution_model`, non-mutating, execution unsupported, and referencing binding owner |
| CLI/API args | still non-authoritative / invalid as execution owner |
| dry-run JSON/model output | evidence only |
| activation/materialization | unchanged; no execution semantics added |
| source events/deliveries/outcomes | invalid paths preserved |
| same-run outbox replay | invalid path preserved |
| runtime bus/pipeline/receipts/dead letters/idempotency/follow-ups | future consumers, not moved in this slice |
| timers/routes/sessions/turns/source-advanced/contract-swap/UI | split blockers / later consumers |

### Old Paths Now Invalid / Non-Authoritative / Surviving

- CLI/API `--contracts` remains input/proof only, not execution truth.
- Dry-run JSON remains evidence only.
- Route/frontier helpers remain admission evidence only.
- Source `events`, source `event_deliveries`, source outcomes, and same-run outbox replay remain invalid as selected-contract fork execution closure.
- No handler execution, fork-local event/delivery writes, receipts, dead letters, idempotency, emitted follow-ups, timers, routes, sessions, turns, source-advanced, contract-swap, Builder/dashboard/API mutation was introduced.

### Broader Failure Class / Symptom Shape

The issue was not treated as symptom-shaped after the #588 gate. The broader class is full mutating selected-contract timestamp-fork execution. This PR closes only the approved first-slice admission/context class and leaves the broader parent open.

Achieved closure level:

- chosen first-slice failure class eliminated.

Parent remains open:

- yes, #588 remains open for full mutating selected-contract execution.

Remaining child-slice tail estimate:

- unchanged by implementation: fork-local handler context, fork-local events/deliveries, receipts/dead letters/idempotency, emitted follow-ups, #575 non-agent policy, #573 sessions/turns, timers/routes, source-advanced policy, contract-swap boot/resume, and UI/API enablement remain future work.

### Sibling Contexts Checked

- #590 / PR #592 durable binding owner.
- #577 / #578 frontier admission.
- #585 / #586 execution model.
- #570 safe-agent replay.
- #573 session/turn blockers.
- #575 non-agent delivery policy.
- Runtime bus, pipeline, receipt, dead-letter, idempotency, emitted follow-up surfaces.
- CLI materialize/dry-run/activate behavior.
- Watchlist nodes `timestamp_fork_replay_resume_ownership` and `delivery_and_replay_ownership`.

### Generic / Focused Proof Created

Created `internal/runtime/runforkexecution/admission.go` and focused tests proving:

- binding is read through the canonical store reader by fork `run_id`.
- missing binding fails closed.
- unavailable selected source fails closed.
- source workflow mismatch fails closed.
- non-canonical frontier/model evidence fails closed.
- admission remains non-mutating and execution unsupported.
- invalid paths and split blockers are preserved from the model.

### Watchlist Decision

Updated `docs/watchlists/runtime-operations.yaml`:

- added #594 under `timestamp_fork_replay_resume_ownership`.
- added the runtime-side selected-contract execution admission/context manifestation.
- did not map this PR to `delivery_and_replay_ownership` because no delivery-write boundary is implemented here.

### Architecture Feedback / Disposition

Architecture smell remains the known parent: selected-contract fork execution still needs a future mutating owner to coordinate pipeline, bus, fork-local event/delivery persistence, receipts, dead letters, idempotency, emitted follow-ups, and source-outcome policy.

Disposition:

- attach to existing trackers #588/#564/#549 and `timestamp_fork_replay_resume_ownership`.
- no new architecture issue needed from this PR because the approved first-slice gap is now closed and the remaining obligations are already tracked.

### Manifestation Coverage

| Manifestation | Status | Exact proof |
| --- | --- | --- |
| runtime admission owner missing | reproduced and fixed | `go test ./internal/runtime/runforkexecution -count=1` |
| future execution could bypass durable binding | reproduced and fixed for this slice | admission requires `RequireRunForkSelectedContractBinding(ctx, forkRunID)`; focused test asserts fork id read |
| missing binding could fall back to CLI/model state | reproduced and fixed | missing binding test fails closed; no fallback path |
| unavailable selected source | reproduced and fixed | nil selected source test fails closed |
| stale/incompatible selected source identity | reproduced and fixed | workflow version mismatch test fails closed |
| non-canonical frontier/model evidence | reproduced and fixed | canonical evidence test fails closed |
| mutating `--contracts` path | execution-proven still blocked | focused `cmd/swarm` proof below |
| handler execution / event-delivery writes / outcomes | split / escalated as separate class | no implementation added; #588 remains open |
| #573/#575/timers/routes/source-advanced/contract-swap/UI | split / escalated as separate class | blockers/watchlist preserved |

### Required Supported-Surface Proof Actually Run

Focused:

- `go test ./internal/runtime/runforkexecution -count=1`
- `go test ./cmd/swarm -run 'TestRunForkCommand_(ContractsRemainNonExecuting|MaterializeOnlyUsesCanonicalStoreOwnerJSON|ActivateUsesCanonicalStoreOwnerJSON|DryRunContractsAddsContractFrontierAdmissionJSON)' -count=1`
- `go test ./internal/store -run 'TestRunForkSelectedContractBinding|TestRunForkMaterializer_CreatesPausedForkRunAndSnapshotWithoutResuming|TestGeneratePlatformTableDDLs' -count=1`

Relevant/full:

- `go test ./internal/runtime/... -count=1`
- `go test ./internal/store -count=1`
- `go test ./cmd/swarm -count=1`
- `go test ./... -count=1`

Closure claim: #594 first-slice failure class is closed. #588 is not closed.

File: internal/runtime/runforkexecution/admission.go [hunk 1]
```diff
@@ -0,0 +1,190 @@
...
+	}
+	if strings.TrimSpace(model.FutureExecutionOwner) != store.RunForkSelectedContractExecutionOwner {
+		return fmt.Errorf("selected-contract execution admission model must point to %s; got %q", store.RunForkSelectedContractExecutionOwner, model.FutureExecutionOwner)
+	}
+	if !model.NonMutating || model.ExecutionSupported {
+		return fmt.Errorf("selected-contract execution admission requires non-mutating unsupported execution model")
+	}
+	if strings.TrimSpace(model.AdmissionOwner) != frontier.Owner {
+		return fmt.Errorf("selected-contract execution admission model admission owner mismatch: got %q want %q", model.AdmissionOwner, frontier.Owner)
+	}
```
[the remark this answers is not in the document]
Comment (yazzaoui) [reply internal/runtime/runforkexecution/admission.go:204]: Addressed in `30486101`. `BuildSelectedContractExecutionAdmission` now fails closed unless the model frontier count and copied frontier events exactly match the supplied `FrontierAdmission`, and `TestBuildSelectedContractExecutionAdmissionFailsClosedOnStaleModelFrontier` covers the stale-model case. Re-ran focused/relevant suites plus `go test ./... -count=1`.

**Comment by yazzaoui** [comment]
Review/CI follow-up:

- Addressed the unresolved review thread in `30486101` by rejecting stale/incompatible execution models whose frontier count/events do not match the supplied `FrontierAdmission`.
- Added `TestBuildSelectedContractExecutionAdmissionFailsClosedOnStaleModelFrontier`.
- Re-ran:
  - `go test ./internal/runtime/runforkexecution -count=1`
  - `go test ./cmd/swarm -run 'TestRunForkCommand_(ContractsRemainNonExecuting|MaterializeOnlyUsesCanonicalStoreOwnerJSON|ActivateUsesCanonicalStoreOwnerJSON|DryRunContractsAddsContractFrontierAdmissionJSON)' -count=1`
  - `go test ./internal/store -run 'TestRunForkSelectedContractBinding|TestRunForkMaterializer_CreatesPausedForkRunAndSnapshotWithoutResuming|TestGeneratePlatformTableDDLs' -count=1`
  - `go test ./internal/runtime/... -count=1`
  - `go test ./... -count=1`

CI note: the GitHub Actions job is failing before checkout/test steps. The check-run annotation says: `The job was not started because recent account payments have failed or your spending limit needs to be increased. Please check the 'Billing & plans' section in your settings`. No code-level CI log is available for this run.

File: internal/runtime/runforkexecution/admission.go [hunk 2]
```diff
@@ -0,0 +1,197 @@
...
+			return fmt.Errorf("selected-contract execution admission binding missing %s", label)
+		}
+		if _, err := uuid.Parse(strings.TrimSpace(value)); err != nil {
+			return fmt.Errorf("selected-contract execution admission binding %s must be a UUID: %w", label, err)
+		}
+	}
+	return validateSelectedContractSelection("binding", binding.ContractSelection)
+}
+
+func validateSelectedContractExecutionSource(binding store.RunForkSelectedContractBinding, source semanticview.Source) error {
```
Comment (yazzaoui) [inline internal/runtime/runforkexecution/admission.go]: Blocking: this still does not validate the selected source against the full durable binding identity. `BuildSelectedContractExecutionAdmission` accepts a caller-supplied `SelectedSource`, and `validateSelectedContractExecutionSource` only compares workflow name/version. The binding identity also includes `contracts_root`, and the spec/#594 require loading and validating the selected semantic source from that binding identity before execution admission. As written, a caller can pass a source from a different root with the same workflow name/version and still receive an admission context, so CLI/API/local source choice can remain part of execution truth. Please make the admission owner load or otherwise verify the source by the binding’s contract selection, including `contracts_root`, and add a negative test for same name/version from the wrong root.

**Review by yazzaoui — COMMENTED** [review]
Blocking review finding: the PR closes the stale-frontier inline comment at current head, and the non-mutating owner/model/frontier accounting is mostly aligned with #594/#588. However, the selected semantic source is still caller-supplied and only matched by workflow name/version, so the admission owner does not actually prove the source was loaded from the durable binding identity before future execution. This is closure-bearing for this first slice because #594 and the spec require binding-identity source validation to fail closed before any selected-contract execution mutation.

**Comment by yazzaoui** [comment]
Checklist review for #595:
- Guideline checks: pre-audit gate verified via #588/#594; spec section cited and inspected; issue category is correct as first-slice failure-class/runtime semantic work; existing stale-frontier inline comment verified obsolete on current head.
- Blocking finding: selected-source validation is not bound to full durable binding identity because `SelectedSource` is caller-supplied and only workflow name/version are compared; `contracts_root` is not loaded/verified from `store.run_fork.selected_contract_binding`.
- Residual risk: future selected-contract execution can still be admitted from a source chosen outside the durable binding, preserving CLI/API/local source truth in the admission seam.
- Risk level: high for the first-slice closure claim.
- Closure level: not achieved; non-mutating model/frontier accounting is improved, but selected-contract execution admission from durable binding remains incomplete.
- Merge recommendation: do not merge until admission loads or verifies source identity from the durable binding, including wrong-root negative proof.
- Watchlist decision: no new node; remains mapped to `timestamp_fork_replay_resume_ownership` with #594.

File: internal/runtime/runforkexecution/admission.go (hunk 2 above)
Comment (yazzaoui) [reply internal/runtime/runforkexecution/admission.go]: Addressed in `0e3a1b22`. The admission request no longer accepts a caller-supplied `SelectedSource`; it now requires a selected-source loader and calls it with the durable binding's full `ContractSelection` loaded by fork `run_id`. The returned source is accepted only if its loaded selection matches the binding across mode, `contracts_root`, workflow name, and workflow version. Added `TestBuildSelectedContractExecutionAdmissionFailsClosedOnWrongContractsRoot` for the same name/version wrong-root case, and the consuming test now asserts the loader receives the binding selection.

**Comment by yazzaoui** [comment]
Review follow-up for the selected-source binding blocker:

- Pushed `0e3a1b22` (`fix: bind fork execution admission source root`).
- `BuildSelectedContractExecutionAdmission` no longer accepts caller-supplied selected source truth.
- Admission now loads selected-source evidence through `LoadRunForkSelectedContractSource(ctx, binding.ContractSelection)` after reading `store.run_fork.selected_contract_binding` by fork `run_id`.
- Source validation now fails closed unless the loaded selection matches the durable binding across mode, `contracts_root`, workflow name, and workflow version.
- Added wrong-root negative proof: `TestBuildSelectedContractExecutionAdmissionFailsClosedOnWrongContractsRoot`.

Proof rerun:
- `go test ./internal/runtime/runforkexecution -count=1`
- `go test ./internal/runtime/... -count=1`
- `go test ./cmd/swarm -run 'TestRunForkCommand_(ContractsRemainNonExecuting|MaterializeOnlyUsesCanonicalStoreOwnerJSON|ActivateUsesCanonicalStoreOwnerJSON|DryRunContractsAddsContractFrontierAdmissionJSON)' -count=1`
- `go test ./internal/store -run 'TestRunForkSelectedContractBinding|TestRunForkMaterializer_CreatesPausedForkRunAndSnapshotWithoutResuming|TestGeneratePlatformTableDDLs' -count=1`
- `go test ./... -count=1`

CI note remains unchanged unless the newly queued run starts: the previous GitHub Actions failure was pre-checkout billing/spending-limit failure, not a test failure.

**Comment by yazzaoui** [comment]
CI update after latest head `0e3a1b22`: the new Actions run also failed before test execution. Check-run annotation is the same billing/spending-limit failure:

`The job was not started because recent account payments have failed or your spending limit needs to be increased. Please check the 'Billing & plans' section in your settings`

No code-level CI log is available.

**Review by yazzaoui — COMMENTED** [review]
Substantive re-review for #595 at `0e3a1b229afca844611684076f44eb544e06d100`:

No blocking findings remain.

The prior selected-source binding blocker is obsolete on current head. `BuildSelectedContractExecutionAdmission` no longer accepts caller-supplied `SelectedSource`; it requires a `SelectedContractSourceLoader`, calls `LoadRunForkSelectedContractSource(ctx, binding.ContractSelection)` after reading `store.run_fork.selected_contract_binding` by fork `run_id`, and validates loaded selection against the durable binding across mode, `contracts_root`, workflow name, and workflow version. The wrong-root negative test covers the same name/version mismatch class.

The earlier stale-frontier inline comment is also obsolete on current head: model frontier count/events are checked against the supplied frontier admission, and the stale-model test covers that path.

Spec compliance assessment: acceptable for the approved #594 first slice. Failure-class coverage assessment: the non-mutating `runtime.run_fork.selected_contract_execution_admission` class is closed. Concept-wide drift-reduction assessment: CLI/API args and local model output are no longer execution truth for this admission seam. Closure level: first-slice failure class eliminated; #588 remains open for mutating selected-contract execution.

Residual risk: GitHub Actions is still failing before test execution because of the account billing/spending-limit annotation, so merge confidence rests on local proof until CI can run. No new follow-up issue is needed; remaining mutating execution work is already tracked by #588/#564/#549 and the existing fork watchlist.

**Comment by yazzaoui** [comment]
Checklist review for #595 re-review:
- Guideline checks: PR description, issue #594, parent gate #588, PR comments, inline comments, spec/watchlist updates, and current head patch inspected.
- Prior blockers: selected-source binding/root validation blocker verified obsolete; stale-frontier model blocker verified obsolete.
- Issue framing: correct as approved first slice for `runtime.run_fork.selected_contract_execution_admission`; broad #588 mutating execution remains open.
- Spec compliance: acceptable for the cited selected-contract execution admission section and #594 boundary.
- Residual risk: CI is externally blocked before checkout/test by billing/spending-limit failure; no code-level CI log is available.
- Risk level: medium, due only to unavailable hosted CI; local proof is strong.
- Closure level: #594 first-slice failure class eliminated; no claim for full mutating selected-contract fork execution.
- Merge recommendation: merge once the team accepts local proof despite CI billing blockage, or rerun CI after billing is fixed.
- Watchlist decision: no new node; #594 remains mapped to `timestamp_fork_replay_resume_ownership`.
